src/test/scala/cromwell/binding/expression/TypeEvaluatorSpec.scala

```diff
@@ -0,0 +1,405 @@
+    (WdlIntegerType, "%", WdlFloatType, WdlFloatType),
+    (WdlIntegerType, "==", WdlIntegerType, WdlBooleanType),
+    (WdlIntegerType, "==", WdlFloatType, WdlBooleanType),
+    (WdlIntegerType, "!=", WdlIntegerType, WdlBooleanType),
+    (WdlIntegerType, "!=", WdlFloatType, WdlBooleanType),
+    (WdlIntegerType, "<", WdlIntegerType, WdlBooleanType),
+    (WdlIntegerType, "<", WdlFloatType, WdlBooleanType),
+    (WdlIntegerType, "<=", WdlIntegerType, WdlBooleanType),
+    (WdlIntegerType, "<=", WdlFloatType, WdlBooleanType),
+    (WdlIntegerType, ">", WdlIntegerType, WdlBooleanType),
+    (WdlIntegerType, ">", WdlFloatType, WdlBooleanType),
+    (WdlIntegerType, ">=", WdlIntegerType, WdlBooleanType),
+    (WdlIntegerType, ">=", WdlFloatType, WdlBooleanType),
+    (WdlFloatType, "+", WdlIntegerType, WdlFloatType),
+    (WdlFloatType, "+", WdlFloatType, WdlFloatType),
+    (WdlFloatType, "+", WdlStringType, WdlStringType),
+    (WdlFloatType, "-", WdlIntegerType, WdlFloatType),
+    (WdlFloatType, "-", WdlFloatType, WdlFloatType),
+    (WdlFloatType, "*", WdlIntegerType, WdlFloatType),
+    (WdlFloatType, "*", WdlFloatType, WdlFloatType),
+    (WdlFloatType, "/", WdlIntegerType, WdlFloatType),
+    (WdlFloatType, "/", WdlFloatType, WdlFloatType),
+    (WdlFloatType, "%", WdlIntegerType, WdlFloatType),
+    (WdlFloatType, "%", WdlFloatType, WdlFloatType),
+    (WdlFloatType, "==", WdlIntegerType, WdlBooleanType),
+    (WdlFloatType, "==", WdlFloatType, WdlBooleanType),
+    (WdlFloatType, "!=", WdlIntegerType, WdlBooleanType),
+    (WdlFloatType, "!=", WdlFloatType, WdlBooleanType),
+    (WdlFloatType, "<", WdlIntegerType, WdlBooleanType),
+    (WdlFloatType, "<", WdlFloatType, WdlBooleanType),
+    (WdlFloatType, "<=", WdlIntegerType, WdlBooleanType),
+    (WdlFloatType, "<=", WdlFloatType, WdlBooleanType),
+    (WdlFloatType, ">", WdlIntegerType, WdlBooleanType),
+    (WdlFloatType, ">", WdlFloatType, WdlBooleanType),
+    (WdlFloatType, ">=", WdlIntegerType, WdlBooleanType),
+    (WdlFloatType, ">=", WdlFloatType, WdlBooleanType),
+    (WdlStringType, "+", WdlIntegerType, WdlStringType),
+    (WdlStringType, "+", WdlFloatType, WdlStringType),
+    (WdlStringType, "+", WdlStringType, WdlStringType),
+    (WdlStringType, "==", WdlStringType, WdlBooleanType),
+    (WdlStringType, "!=", WdlStringType, WdlBooleanType),
+    (WdlStringType, "<", WdlStringType, WdlBooleanType),
+    (WdlStringType, "<=", WdlStringType, WdlBooleanType),
+    (WdlStringType, ">", WdlStringType, WdlBooleanType),
+    (WdlStringType, ">=", WdlStringType, WdlBooleanType),
+    (WdlFileType, "+", WdlStringType, WdlFileType),
+    (WdlFileType, "+", WdlFileType, WdlFileType),
+    (WdlFileType, "==", WdlStringType, WdlBooleanType),
+    (WdlFileType, "==", WdlFileType, WdlBooleanType),
+    (WdlFileType, "!=", WdlStringType, WdlBooleanType),
+    (WdlFileType, "!=", WdlFileType, WdlBooleanType),
+    (WdlFileType, "<=", WdlStringType, WdlBooleanType),
```

Comment:
With current semantics it'll return `WdlBoolean.False`, because `<` is not defined on but `==` is defined, so it'd return the result of `WdlFile("/dev/null") == WdlString("foo")`.  In terms of binary operators, I tried to keep String and File relatively close in semantics.

Does that seem weird to you?  Any other suggestions?